src/ast/nodes/VariableDeclaration.ts

```diff
@@ -73,101 +60,98 @@ export default class VariableDeclaration extends NodeBase {
 		);
 	}
 
-	render (code: MagicString, options: RenderOptions) {
-		const treeshake = this.module.graph.treeshake;
-
-		let shouldSeparate = false;
-		let separator;
-
-		if (this.scope.isModuleScope && !forStatement.test(this.parent.type)) {
-			shouldSeparate = true;
-			separator = getSeparator(this.module.code, this.start);
+	render (code: MagicString, options: RenderOptions, nodeRenderOptions: NodeRenderOptions = {}) {
+		const separatedNodes = getCommaSeparatedNodesWithSeparators(
+			this.declarations,
+			code,
+			this.start + this.kind.length,
+			this.end - (code.original[this.end - 1] === ';' ? 1 : 0)
+		);
+		let renderedContentEnd;
+		if (/\n\s*$/.test(code.slice(this.start, separatedNodes[0].start))) {
+			renderedContentEnd = this.start + this.kind.length;
+		} else {
+			renderedContentEnd = separatedNodes[0].start;
 		}
-
-		let c = this.start;
-		let empty = true;
-		let lastSystemWrap = false;
-		let curSystemWrap = false;
-
-		for (let i = 0; i < this.declarations.length; i += 1) {
-			const declarator = this.declarations[i];
-
-			const prefix = empty ? '' : separator; // TODO indentation
-
-			if (isIdentifier(declarator.id)) {
-				const variable = this.scope.findVariable(declarator.id.name);
-
-				const isExportedAndReassigned = variable.safeName && variable.safeName.indexOf('.') !== -1 && variable.exportName && variable.isReassigned;
-
-				if (options.systemBindings && variable.exportName && declarator.init) {
-					code.prependLeft(declarator.init.start, `exports('${variable.exportName}', `);
-					curSystemWrap = true;
-				}
-
-				if (isExportedAndReassigned) {
-					if (declarator.init) {
-						if (shouldSeparate) code.overwrite(c, declarator.start, (lastSystemWrap ? ')' : '') + prefix);
-						c = declarator.end;
-						empty = false;
-					}
-				} else if (!treeshake || variable.included) {
-					if (shouldSeparate)
-						code.overwrite(c, declarator.start, `${lastSystemWrap ? ')' : ''}${prefix}${this.kind} `); // TODO indentation
-					c = declarator.end;
-					empty = false;
+		let lastSeparatorPos = renderedContentEnd - 1;
+		code.remove(this.start, lastSeparatorPos);
+		let isInDeclaration = false;
+		let hasRenderedContent = false;
+		let separatorString = '', leadingString, nextSeparatorString;
+		for (let { node, start, separator, end } of separatedNodes) {
+			if (!node.included || (isIdentifier(node.id) && isReassignedPartOfExportsObject(node.id.variable) && node.init === null)) {
+				code.remove(start, end);
+				continue;
+			}
+			leadingString = '';
+			nextSeparatorString = '';
+			if (isIdentifier(node.id) && isReassignedPartOfExportsObject(node.id.variable)) {
```

Comment:
Perhaps `isReassignedAsExportsMember`? `Object` as a term isn't descriptive enough for me to immediately think of a member expression form without further prompting.

Reply:
👍